RIOT-OS/RIOT
# boards: add support for ESP32-S3 WT32 SC01 Plus board

### Contribution description

This PR provides support for the [Wireless Tag WT32-SC01 Plus with ESP32-S3](http://en.wireless-tag.com/product-item-26.html) board. The board has the following main features:

- ESP32-S3 SoC
- 16 MB Flash
- 2 MB QSPI RAM
- 3.5" LCD Display 480 x 320 with ST7796UI
- Capacitive Touch Panel with FT6336U
- SD Card SPI mode
- USB Type-C

**Please note**: The ST7796UI display uses the MCU8080 8-bit parallel interface which require the changes in PR #19915.

### Testing procedure

1. `tests/drivers/st77xx` should work on top of PR #19915.
   ```
   BOARD=esp32s3-wt32-sc01-plus make -j8 -C tests/drivers/st77xx flash
   ```
2. `tests/drivers/ft5x06` should work with `stdio_uart` and the debugging tool.
    ```
    USEMODULE=stdio_uart BOARD=esp32s3-wt32-sc01-plus make -j8 -C tests/drivers/ft5x06/ flash term
    ```
    ```
    main(): This is RIOT! (Version: 2023.10-devel-245-g55bd7-boards/esp32s3-wt32-sc01-plus)
    FT5x06 test application

    +------------Initializing------------+
    Initialization successful
    1 touch detected
    Touch 1 - X: 177, Y:96
    Touch 1 - X: 177, Y:96
    Touch 1 - X: 177, Y:96
    Touch 1 - X: 177, Y:97
    Touch 1 - X: 178, Y:99
    Touch 1 - X: 180, Y:102
    Touch 1 - X: 184, Y:106
    Touch 1 - X: 190, Y:110
    Touch 1 - X: 197, Y:116
    Touch 1 - X: 204, Y:122
    Released!
    ```

### Issues/PRs references
